os_migrate/plugins/module_utils/workload_common.py

```diff
@@ -148,6 +116,40 @@ def __init__(self, openstack_connection, conversion_host_id, ssh_key_path,
         # Ports chosen for NBD export
         self.claimed_ports = []
 
+    def check_process(self, pid):
+        """Check whether pid exists in the current process table."""
+        if pid is None:
+            return False
+        try:
+            self.shell.cmd_val(['kill', '-0', str(pid)])
+        except subprocess.CalledProcessError:
+            return False
+        else:
+            return True
+
+    def check_and_cleanup_lockfiles(self):
+        """
+        Removes lockfiles found in specific directories.
+        If a lockfile is not being used by a process,
+        it is either deleted or not being used.
+        """
+        # Remove specific lockfiles
+        for lockfile in [ATTACH_LOCK_FILE_SOURCE, ATTACH_LOCK_FILE_DESTINATION, PORT_LOCK_FILE]:
+            try:
+                # Remove specific lockfiles
+                for lockfile in [ATTACH_LOCK_FILE_SOURCE, ATTACH_LOCK_FILE_DESTINATION, PORT_LOCK_FILE]:
+                # Use self.shell.cmd_out to run the following command on the conversion host
+                pid = self.shell.cmd_out(f'cat {lockfile}.lock | grep -Po "(?<=^PID )\d+"')
+                if self.check_process(pid):
+                    continue                
+                else:
+                    # The lockfile is not being used by a process, so we can remove it
+                    # self.shell.cmd_out(f'rm {lockfile}.lock')
```

Comment:
@jistr I assume we're running as root user to begin with, if so I'll uncomment this command if not the command below will execute - users will need to run --ask-pass to get this to work so I can see a problem here